src/job_cache/job_cache.cpp

```diff
@@ -1145,10 +1145,11 @@ wcl::optional<MatchingJob> Cache::read(const FindJobRequest &find_request) {
       mkdir_all(pair.second.begin(), pair.second.end());
 
       // Lastly make the symlink
-      if (symlink(pair.first.c_str(), output_symlink.value.c_str()) == -1) {
-        log_fatal("symlink(%s, %s): %s", pair.first.c_str(), output_symlink.value.c_str(),
-                  strerror(errno));
+      std::string tmp_link = rng.unique_name();
```

Comment:
It needs to be adjacent to the output_symlink location so this should be `output_symlink.value + "." + rng.unique_name()` that also makes debugging easier if we ever find one of those files lying around.